pkg/controller/v1beta1/inferenceservice/reconcilers/knative/ksvc_reconciler.go

```diff
@@ -85,6 +85,13 @@ func createKnativeService(componentMeta metav1.ObjectMeta,
 		annotations[constants.MinScaleAnnotationKey] = fmt.Sprint(*componentExtension.MinReplicas)
 	}
 
+	// The larger of min-scale and initial-scale is chosen as the initial target scale for a knative Revision.
+	// When min-scale is 0, set initial-scale to 0 so the created knative revision has an initial target scale of 0.
+	// Configuring scaling for knative: https://knative.dev/docs/serving/autoscaling/scale-bounds/#initial-scale
+	if annotations[constants.MinScaleAnnotationKey] == "0" {
+		annotations[constants.InitialScaleAnnotationKey] = "0"
+	}
+
```

Comment:
to move this check into if..else above
```suggestion

	if componentExtension.MinReplicas == nil {
		annotations[constants.MinScaleAnnotationKey] = strconv.Itoa(constants.DefaultMinReplicas)
	} else {
		annotations[constants.MinScaleAnnotationKey] = strconv.Itoa(*componentExtension.MinReplicas)
		// The larger of min-scale and initial-scale is chosen as the initial target scale for a knative Revision.
		// When min-scale is 0, set initial-scale to 0 so the created knative revision has an initial target scale of 0.
		// Configuring scaling for knative: https://knative.dev/docs/serving/autoscaling/scale-bounds/#initial-scale
		if *componentExtension.MinReplicas < constants.DefaultMinReplicas {
			annotations[constants.InitialScaleAnnotationKey] = "0"
		}
	}
```

Reply:
btw, why we do not check nil pointer on `if componentExtension.MaxReplicas != 0 {... ` line 95 ?

Reply:
@zdtsw MaxReplicas value is not a pointer

Reply:
@zdtsw I'm not sure I see the need to nest this check inside the other if else block. Could you elaborate?

Additionally, I don't think there is a need to check `if *componentExtension.MinReplicas < constants.DefaultMinReplicas` instead of `if annotations[constants.MinScaleAnnotationKey] == "0"`. If I understand the logic correctly, we already have a check [here](https://github.com/opendatahub-io/kserve/blob/master/pkg/apis/serving/v1beta1/component.go#L177) that will throw an error if MinReplicas < 0. This logic could also introduce a problem if the constants.DefaultMinReplicas value is changed to be > 0.